lib/state.js

```diff
@@ -825,7 +825,111 @@ class State {
     }
   }
 
-  async logCloudWatchEvent({region, id, state, generated}, client) {
+  /**
+   * Log a request to AWS services
+   */
+  async logAWSRequest({
+    region,
+    requestId,
+    duration,
+    method,
+    service,
+    error,
+    called,
+    code,
+    message,
+    workerType,
+    az,
+    instanceType,
+    imageId,
+  }) {
+    assert(typeof region === 'string');
+    assert(typeof requestId === 'string');
+    assert(typeof duration === 'number');
+    assert(typeof method === 'string');
+    assert(typeof service === 'string');
+    assert(typeof error === 'boolean');
+    assert(typeof called === 'object');
+    assert(called.constructor.name === 'Date');
+
+    let columns = ['region', 'requestId', 'duration', 'method', 'service', 'error', 'called'];
+
+    let values = [
+      region,
+      requestId,
+      duration,
+      method,
+      service,
+      error,
+      called,
+    ];
+
+    if (code || message || error) {
+      assert(typeof code === 'string');
+      assert(typeof message === 'string');
+      assert(error);
```

Comment:
nope, because you could have truthy `code`, but falsy `error`, which is what this is checking.  It's saying that if you have either a code or a message or it's an error, that you must have a code, a message and you must set error to true